backend/api/routes/users.js

```diff
@@ -24,4 +24,29 @@ router.route("/me").get(authenticateMiddleware, async (req, res) => {
   }
 });
 
+router.route("/get-all").get(authenticateMiddleware, async (req, res) => {
+  try {
+    //putting client that we attached to request in a variable for ease of use
+    const supabase = req.supabase;
+
+    //fetch all profile rows
+    const { data: streamers, error } = await supabase
```

Comment:
You could add a TODO item here to know where you will be adding the discover algorithm in the future